core/commands/SubprojectAddCommand.php

```diff
@@ -0,0 +1,146 @@
+<?php
+# MantisBT - A PHP based bugtracking system
+
+# MantisBT is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# MantisBT is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with MantisBT.  If not, see <http://www.gnu.org/licenses/>.
+
+require_api( 'constant_inc.php' );
+require_api( 'config_api.php' );
+require_api( 'helper_api.php' );
+require_api( 'project_api.php' );
+require_api( 'project_hierarchy_api.php' );
+
+use Mantis\Exceptions\ClientException;
+
+/**
+ * A command that adds a project subproject.
+ */
+class SubprojectAddCommand extends Command {
+
+	/**
+	 * @var integer
+	 */
+	private $project_id;
+
+	/**
+	 * @var integer
+	 */
+	private $subproject_id;
+
+	public function get_subproject_id() {
+		return $this->subproject_id;
+	}
+
+	/**
+	 * Constructor
+	 *
+	 * $p_data['query'] is expected to contain:
+	 * - project_id (integer)
+	 *
+	 * $p_data['payload'] is expected to contain:
+	 * - project (array)
+	 * - inherit_parent (bool)
+	 *
+	 * @param array $p_data The command data.
+	 */
+	function __construct( array $p_data ) {
+		parent::__construct( $p_data );
+	}
+
+	private function get_subproject_project_id( $p_project ) {
```

Comment:
Use `mci_get_project_id()` similar to `IssueAddCommand`.  Also this method is missing phpdocs, but it should be removed anyway.

Reply:
The only reason I added that private, is to distinguish neither `id` nor `name` are set from
only `name` is set but that project doesn't exist, and reflect that to the caller.

`IssueAddCommand` BTW raises 'Project not specified' in both cases, which is at least questionable for the latter.

Reply:
I would rather improve `mc_get_project_id()` by adding an optional parameter for default project id ($p_default = ALL_PROJECTS).  You can call it with default value `false` and that will give you the distinction.  We can eventually update other callers to do the same.